### ISSUE TITLE
rewriting the notification module settings, reorganising the module and adding additional instructions

### ISSUE THREAD

**Issue by hermithome** [creesch/reddit-moderator-toolbox#447]
With the help of psdtwk, I've taken a stab at rewriting the notification module settings so that they clearly state what they do.

Here's my initial stab at what it should look like. I've reorganised the module by diving it into two sections: counter settings and notification settings. Both sections include a short para at the top, explaining how they work, and introducing the terminology used in the settings. And I created three sub-sections for notification settings into three sections: personal messages, modmail and queues. Visually I imagined a little header and the section being indented.

I've also reworded most of the individual setting options. I tried to strike a balance between brevity and clarity. 

I forked the page so I could rearrange stuff and add in language where I think it should go, just to give a good sense of the changes I want to see. It is totally non-functional code though, as I just added text in wherever I thought it should go, I didn't even comment it out. Hopefully though, it gives a good sense of the changes I think should be made for user understandability.

https://github.com/hermithome/reddit-moderator-toolbox/blob/patch-1/modules/notifier.js

**Comment by creesch** [comment]
As I said earlier this is no functional code. I am not sure we want to include that much text on setting pages themselves. A lot of people when looking at settings will tl;dr if there is a lot of text on the assumption that it must be complicated stuff. 

So personally I'd like to keep documentation and functionality seperated in that regard. 

That doesn't mean we can't look at the current labels and clarify those a bit. However [this for example](https://github.com/hermithome/reddit-moderator-toolbox/blob/patch-1/modules/notifier.js#L111) is way too much text.

**Comment by creesch** [comment]
<agentlame> creesch: I think there may be some merit to her idea of grouping settings.
<creesch> Grouping settings is fine 
<creesch> I don't want documentation text on the page though 
<creesch> At least not in the way proposed 
<agentlame> Yeah, it's a bit verbose.
<agentlame> But adjusting some of the language might be a good idea.
<creesch> it would be neat if we had the ability add an (i) icon that on hover shows a little tooltip. But honestly that is also a TON of work since people would expect it on all settings.
<agentlame> It's an idea for the future.

**Comment by hermithome** [comment]
psdtwk and I discussed the same things actually.

re documentation: 
Most people don't use the documentation, and many aren't aware that it's there. Yes, that ? links to it, but many people don't notice it, and the tooltip text for it just says "javascript;;" which is enough to put off a lot of people. I suggest either making the ? red, and writing in an actual tooltip that says that it opens the corresponding documentation page, or adding a line at the bottom of each module linking to the documentation page. One line of text, just above the bottom border, with a pinkish background should do the trick.

I'm must more inclined then the average user to seek out and pour over documentation. And I went to the documentation for toolbox, once, and only once: when I first installed it. And I went there from the sub, I had no clue that the modules linked to each documentation page. At that time, the documentation that existed was small, and fairly confusing. And it didn't provide the help I needed. I didn't look at the documentation again until I started writing it.

re paring down and tooltips:
I do agree that most of the instructions could be pared down a lot. I think ideally, I'd like to see a combination of things: a more obvious link to the documentation, tooltip text, and reorganisation and rewording. I can divide the text I've written into stuff for documentation, primary instruction, and secondary tooltip pretty easily.

Yes, people will expect the tooltips everywhere, but there's no reason we can't start with them in just a few modules. Besides, writing tooltip text is fairly easy, and I could do it for the other modules as well. I started with the notification module because it was, IMO, the most confusing module. Even though I've used notifications for ages, I honestly didn't know how some of those options worked, and neither did psdtwk. That's not good.

The one longer bit that I'd really fight to keep in some manner, is the intro to the toolbar. It includes a lot of vital information that sadly, most people don't know: what the instructions mean by "multireddit", how to edit the multi, how the counters work, and that the counters and notifications are tied. I know a lot of that stuff seems super obvious, but the thing is, it isn't for most people. A lot of people who come to toolbox for the first time have never touched the unmoderated queue, they've never looked at the url or a multi, they don't know about the redditmade "mod" queue and so on.

Being able to include a short explanation in the module I think is pretty important. It could be done through tooltips, but I do think it's pretty important.

**Comment by hermithome** [comment]
just did a quick pass of the file on my fork, commenting out stuff that i think would be better as a tooltip or doc

**Comment by creesch** [comment]
> I suggest either making the ? red

That is not likely to happen

> writing in an actual tooltip that says that it opens the corresponding documentation page,

Good idea. 

>  I'm must more inclined then the average user to seek out and pour over documentation. And I went to the documentation for toolbox, once, and only once: when I first installed it. And I went there from the sub, I had no clue that the modules linked to each documentation page. At that time, the documentation that existed was small, and fairly confusing. And it didn't provide the help I needed. I didn't look at the documentation again until I started writing it.

I am sorry, this is highly anecdotal and hangs together from assumptions based on your own behavior you yourself described as being different from other users. 
Either way, moving it on the page is not going to make people more inclined to read it. it is much more likely to make them avoid the settings altogether since it looks even more complicated now. 

> Yes, people will expect the tooltips everywhere, but there's no reason we can't start with them in just a few modules. Besides, writing tooltip text is fairly easy, and I could do it for the other modules as well. I started with the notification module because it was, IMO, the most confusing module. Even though I've used notifications for ages, I honestly didn't know how some of those options worked, and neither did psdtwk. That's not good.

Yeah, I am just saying that if we implement this it will be a long term goal. Since it is rather labor intensive to get right. We have limited resources at our disposal, time is the main factor here. 



**tl;dr** Basically it comes down to this for me. 

Yes on: 

- Hover tooltip on help button (Which btw in the recent release has been standing out more)
- Long term: better grouping of settings. 
- Longer term: Tooltips next to settings

No on: 

- More text in the settings themselves.

**Comment by creesch** [comment]
Closed the issue since the title does not cover anything that will happen anytime soon in this capacity.

**Comment by hermithome** [comment]
>I am sorry, this is highly anecdotal and hangs together from assumptions based on your own behavior you yourself described as being different from other users.

Actually no, I've done documentation for a lot of projects and something I've continually found is that a large number of users do not seek out documentation. They often need it to be linked prominently. And if documentation is confusing, or partial, they often give up entirely. Sorry, the personal anecdote was really irrelevant to what I was trying to say. And I've trained a number of mods on toolbox. Almost no one noticed the question mark, and none of them clicked it. A grand total of two went to the sub and checked the wiki, and they complained that the documentation was confusing or incomplete.

I've been writing how-to's for toolbox for my mods in subreddit wikis because of this.

I know you like clean UI, and I appreciate that. But I'm not sure how much of a point there is in writing great docs if few people seek them out. I'm hoping you'll think about that, and at least consider linking the documentation in some sort of more prominent manner.

Re groupings and tooltips....whenever you're at a point to do them technically, just poke me. I could probably write tooltips for the whole damn thing in a couple of days, and groupings would take even less time, so just let me know when you need them.

**Comment by agentlame** [comment]
>I've been writing how-to's for toolbox for my mods in subreddit wikis because of this.

I'm confused.  If you're writing how-to's for toolbox why wouldn't you contribute them back to /r/toolbox/wiki?

Obviously you're not obligated to, but it would be of value to all mods.  And it's not like @psdtwk couldn't use the help.

Not to get too far off-topic; you have to remember that creesch and I have a finite amount of free time for TB.  Most of it gets spent adding stuff and fixing stuff.

**Comment by xminusone** [comment]
A separate "how-to" section of the wiki would be cool, and I'd totally be willing to contribute at least my "how to nuke a whole thread's comments" how-to.

**Comment by agentlame** [comment]
I'm gonna reopen this for now.  We don't have proper issues for each item discussed here, and until we do I don't want the valuable parts of this issue lost.

**Comment by psdtwk** [comment]
@hermithome @xminusone 

Each of the official wiki pages should have a **Usage** section, with numbers steps on how to do things. See the "[how to make a Removal Reason for beginners](http://www.reddit.com/r/toolbox/wiki/docs/removalreasons#wiki_beginner_editing)" for an example. It's four steps, click by click.

If you want to help with docs in a huge way, then let's move the discussion to the subreddit and not clog up this separate issue. As agentlame said, I like the help and the fresh eyes.

**Comment by hermithome** [comment]
>I'm confused. If you're writing how-to's for toolbox why wouldn't you contribute them back to /r/toolbox/wiki?

They aren't immediately transferable, they need significant rewrites. Most of the how-to's include highly specific information for each sub, and they only include the tools that I need the other mods to learn how to use.

I'd be happy to write more documentation for toolbox, include how-to guides, but I think that only solves part of them problem. And yeah, @psdtwk is right, that's a conversation for elsewhere. 

I'm happy to write documentation, and tooltips and groupings. But I do still think that documentation needs to be linked to in a more prominent/obvious way, and I think that some of the modules need one line or two or text introducing them. Not huge chunks, but a little. You shouldn't have users coming to a page and not having any idea what anything does, and IME training mods with toolbox, way too many mods look at the notification module and have no idea what anything there is talking about.
